src/Ast/PreviousConnectingVisitor.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+declare(strict_types=1);
+
+namespace staabm\PHPStanDba\Ast;
+
+use PhpParser\Node;
+use PhpParser\NodeVisitorAbstract;
+use function array_pop;
+use function count;
+use function get_class;
+
+final class PreviousConnectingVisitor extends NodeVisitorAbstract
```

Comment:
this impl is pretty similar to `PhpParser\NodeVisitor\NodeConnectingVisitor` minus the `next` attribute.

I think there is room for improvement, but at first I want to get bleedingEdge compat, so we can play with other bleeding edge features unrelated to memory/performance optimizations